ad/resource_active_directory_computer_to_ou.go

```diff
@@ -0,0 +1,113 @@
+package ad
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	ldap "gopkg.in/ldap.v2"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceComputerToOU() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceADComputerToOUCreate,
+		Read:   resourceADComputerToOURead,
+		Delete: resourceADComputerToOUDelete,
+		Schema: map[string]*schema.Schema{
+			"computer_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ou_distinguished_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceADComputerToOUCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ldap.Conn)
+
+	computerName := d.Get("computer_name").(string)
+	OUDistinguishedName := d.Get("ou_distinguished_name").(string)
+	var dnOfComputer string
+	dnOfComputer += "cn=" + computerName + "," + OUDistinguishedName
+
+	log.Printf("[DEBUG] Name of the DN is : %s ", dnOfComputer)
+	log.Printf("[DEBUG] Adding the Computer to the AD : %s ", computerName)
+
+	err := addComputerToAD(computerName, dnOfComputer, client)
+	if err != nil {
+		log.Printf("[ERROR] Error while adding a Computer to the AD : %s ", err)
+		return fmt.Errorf("Error while adding a Computer to the AD %s", err)
+	}
+	log.Printf("[DEBUG] Computer Added to AD successfully: %s", computerName)
+	d.SetId(OUDistinguishedName + "/" + computerName)
+	return nil
+}
+
+func resourceADComputerToOURead(d *schema.ResourceData, meta interface{}) error {
+	log.Println("[ERROR] In Read function")
+	client := meta.(*ldap.Conn)
+
+	computerName := d.Get("computer_name").(string)
+	OUDistinguishedName := d.Get("ou_distinguished_name").(string)
+	var dnOfComputer string
+	dnOfComputer += OUDistinguishedName
+
+	log.Printf("[DEBUG] Name of the DN is : %s ", dnOfComputer)
+	log.Printf("[DEBUG] Searching the Computer from the AD : %s ", computerName)
+
+	searchRequest := ldap.NewSearchRequest(
+		dnOfComputer, // The base dn to search
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		"(&(objectClass=Computer)(cn="+computerName+"))", // The filter to apply
+		[]string{"dn", "cn"}, // A list attributes to retrieve
+		nil,
+	)
+
+	sr, err := client.Search(searchRequest)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Please do not use log.Fatal as it will exit the extension https://golang.org/src/log/log.go?s=9291:9319#L295
so please add a log instead describing the error and return the error

Reply:
Done